pkg/client/unversioned/configdata.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type ConfigDatasNamespacer interface {
+	ConfigDatas(namespace string) ConfigDatasInterface
+}
+
+type ConfigDatasInterface interface {
+	Get(string) (*extensions.ConfigData, error)
+	List(labels.Selector, fields.Selector) (*extensions.ConfigDataList, error)
+	Create(*extensions.ConfigData) (*extensions.ConfigData, error)
+	Delete(string) error
+	Update(*extensions.ConfigData) (*extensions.ConfigData, error)
+	UpdateStatus(*extensions.ConfigData) (*extensions.ConfigData, error)
+	Watch(labels.Selector, fields.Selector, string) (watch.Interface, error)
+}
+
+type configDatas struct {
+	client    *Client
+	namespace string
+}
+
+func newConfigDatas(c *Client, ns string) *configDatas {
+	return &configDatas{
+		client:    c,
+		namespace: ns,
+	}
+}
+
+func (cfg *configDatas) resourceName() string {
+	return "configDatas"
+}
+
+func (cfg *configDatas) Get(name string) (*extensions.ConfigData, error) {
+	result := &extensions.ConfigData{}
+	err := cfg.client.Get().
+		Namespace(cfg.namespace).
+		Resource(cfg.resourceName()).
+		Name(name).
+		Do().
+		Into(result)
+
+	return result, err
+}
+
+func (cfg *configDatas) List(label labels.Selector, field fields.Selector) (*extensions.ConfigDataList, error) {
+	result := &extensions.ConfigDataList{}
+	err := cfg.client.Get().
+		Namespace(cfg.namespace).
+		Resource(cfg.resourceName()).
+		LabelsSelectorParam(label).
+		FieldsSelectorParam(field).
+		Do().
+		Into(result)
+
+	return result, err
+}
+
+func (cfg *configDatas) Create(config *extensions.ConfigData) (*extensions.ConfigData, error) {
+	result := &extensions.ConfigData{}
+	err := cfg.client.Post().
+		Namespace(cfg.namespace).
+		Resource(cfg.resourceName()).
+		Body(config).
+		Do().
+		Into(result)
+
+	return result, err
+}
+
+func (cfg *configDatas) Delete(name string) error {
+	return cfg.client.Delete().
+		Namespace(cfg.namespace).
+		Resource(cfg.resourceName()).
+		Name(name).
+		Do().
+		Error()
+}
+
+func (oldCfg *configDatas) Update(newCfg *extensions.ConfigData) (*extensions.ConfigData, error) {
```

Comment:
the pattern in other update methods seems to be to use the name of the resource for the parameter name.  Go ahead and change it to:

```
func (c *configDatas) Update(config *extensions.ConfigData) (*extensions.ConfigData, error)
```